sympy/series/sequences.py

```diff
@@ -0,0 +1,522 @@
+        return ()
+
+    def coeff(self, i):
+        """Returns the coefficient at point i"""
+        if i < self.start or i > self.stop:
+            raise IndexError("Index %s out of bounds %s" %(i, self.interval))
+        return self._eval_coeff(i)
+
+    def _eval_coeff(self, i):
+        raise NotImplementedError(filldedent(""" The _eval_coeff method should\
+                                             be added to %s to return\
+                                             coefficient so it is available\
+                                             when coeff calls it."""\
+                                             % self.func))
+
+    def _ith_point(self, i, step=None):
+        """
+        Returns the i'th point of a sequence
+        If start point is negative infinity, point is returned from the end.
+        Assumes the first point to be indexed zero.
+
+        Examples
+        =========
+
+        >>> from sympy import oo
+        >>> from sympy.series.sequences import SeqExpr
+
+        bounded
+
+        >>> SeqExpr((1, 2, 3), (-10, 10))._ith_point(0)
+        -10
+        >>> SeqExpr((1, 2, 3), (-10, 10))._ith_point(5)
+        -5
+        >>> SeqExpr((1, 2, 3), (-10, 10, 2))._ith_point(5)
+        0
+
+        End is at infinity
+
+        >>> SeqExpr((1, 2, 3), (0, oo))._ith_point(5)
+        5
+        >>> SeqExpr((1, 2, 3), (0, oo, 2))._ith_point(5)
+        10
+
+        Starts at negative infinity
+
+        >>> SeqExpr((1, 2, 3), (-oo, 0))._ith_point(5)
+        -5
+        >>> SeqExpr((1, 2, 3), (-oo, 0, 2))._ith_point(5)
+        -10
+        """
+        if self.start is S.NegativeInfinity:
+            initial = self.stop
+        else:
+            initial = self.start
+
+        if step == None:
+            if self.start is S.NegativeInfinity:
+                step = -self.step
+            else:
+                step = self.step
+
+        return initial + i*step
+
+    def __getitem__(self, index):
+        if isinstance(index, integer_types):
+            index = self._ith_point(index)
+            return self.coeff(index)
+        elif isinstance(index, slice):
+            start, stop = index.start, index.stop
+            if start == None:
+                start = 0
+            if stop == None:
+                stop = self.length
+            return [self.coeff(self._ith_point(i, index.step)) for i in\
+                               range(start, stop)]
+
+
+class EmptySequence(with_metaclass(Singleton, SeqBase)):
+    """
+    Represents an empty sequence. The empty sequence is available as a
+    singleton as S.EmptySequence.
+
+    Examples
+    ========
+
+    >>> from sympy import S
+    >>> S.EmptySequence
+    EmptySequence()
+    """
+
+    is_iterable = True
+
+    @property
+    def _interval(self):
+        return S.EmptySet
+
+    @property
+    def _length(self):
+        return S.Zero
+
+    def __iter__(self):
+        return iter([])
+
+
+def _parse_interval(interval):
```

Comment:
Well, I took ``Sum`` as a reference. For something like ``Sum(x**2, (x, 0, 5))`` 5 is included, so going by that convention I decided to include the end point as well.